src/Uno.UWP/Storage/Streams/DataReader.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Windows.Storage.Streams
+{
+	public partial class DataReader : IDataReader, IDisposable
+	{
+		private IBuffer _buffer = null;
+		private uint _position = 0;
+
+		public ByteOrder ByteOrder { get; set; }
+
+		public uint UnconsumedBufferLength
+		{
+			get => _buffer.Length - _position;
+		}
+		//public DataReader(global::Windows.Storage.Streams.IInputStream inputStream)
+		//{
+		//	global::Windows.Foundation.Metadata.ApiInformation.TryRaiseNotImplemented("Windows.Storage.Streams.DataReader", "DataReader.DataReader(IInputStream inputStream)");
+		//}
+
+		private void CheckPosition(int size)
+		{
+			if (_buffer.Length >= (_position + size))
+			{
+				return;
+			}
+
+			throw new IndexOutOfRangeException("Windows.Storage.Streams.DataReader - reading past EOF");

```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['System.IndexOutOfRangeException' should not be thrown by user code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5527039)

Reply:
So, what exception should be thrown if pointer is out of range?

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.argumentoutofrangeexception?view=dotnet-uwp-10.0

Reply:
This doc says: "when the value of an argument is outside". Here, we don't have argument out of range. "arguments passed to the method is not null and contains an invalid value" - no, it is not problem with arguments. It is reading past EndOfFile :)

Reply:
But, neverless, only some methods from this implementation would be used, as it is also implemented in previous (not yet merged) PR, #2930.